### PR TITLE
Feature - Generic Create Component

### PR DESCRIPTION
**Pull request by westlywright**
Adds a generic `CruResource` component. 

`CruResource` has 3 states:

Subtype selection: If the resource has subtypes that can be selected, e.g. service cluster ip/external name/ etc, it will show this on create. On edit the user will bypass subtype selection and proceed directly to the form view.

Form Editor: Exposes a define slot which allows the consumer to pass in the form. This was made to be as generic as possible but this could be updated to include the tabbed/tab and build the forms but imo would make the component much less dynamic so I opted for using a slot here where we can pass tabbed/tab down.

Yaml Editor: Exposes a yaml-review action where the user can further modify the resources yaml. Cancelling/back from this state will trigger confirmation modals.

Screens:
Initial State:
Screen Shot 2020-08-03 at 10 58 43-fullpage

Form Edit:
Screen Shot 2020-08-03 at 10 58 52-fullpage

Yaml Edit:
Screen Shot 2020-08-03 at 10 59 32-fullpage

Go Back Modal State:
Screen Shot 2020-08-03 at 10 59 37-fullpage

Cancel Modal State:
Screen Shot 2020-08-03 at 11 22 40-fullpage

Edit Form State:
Screen Shot 2020-08-03 at 11 00 00-fullpage

Edit Preview Yaml State:
Screen Shot 2020-08-03 at 11 00 05-fullpage

### CODE REVIEW

**Comment by deniseschannon** [comment]
Looks fine from my point of view but what happened to labels and annotations?

**Comment by lvuch** [comment]
lgtm on my list is modal styling. note that port rules should be grid not tables. @vincent99 were you going to reposition pagination or have feedback?

**Comment by westlywright** [comment]
@deniseschannon Sorry about that, added them back in. 

Screen Shot 2020-08-03 at 14 14 17-fullpage

File: components/CruResource.vue [hunk 1]
```diff
@@ -0,0 +1,404 @@
...
+
+    selectedSubtype: {
+      type:    String,
+      default: null
+    },
+
+    canCreate: {
+      type:    Boolean,
+      default: false
+    }
```
[the remark this answers is not in the document]
Comment (westlywright) [reply components/CruResource.vue:46]: I'll swap the default to true but this is meant to serve as a validation flag. The name, I realize does not convey this at all and I'll fix that. But yeah the idea is that the consumer can pass down ready flag if we're preventing you from creating via form validation.

**Pull request merged**